numpy/lib/function_base.py

```diff
@@ -3914,74 +3921,57 @@ def _quantile_ureduce_func(a, q, axis=None, out=None, overwrite_input=False,
             "interpolation can only be 'linear', 'lower' 'higher', "
             "'midpoint', or 'nearest'")
 
-    n = np.array(False, dtype=bool)  # check for nan's flag
-    if np.issubdtype(indices.dtype, np.integer):  # take the points along axis
-        # Check if the array contains any nan's
-        if np.issubdtype(a.dtype, np.inexact):
-            indices = concatenate((indices, [-1]))
-
-        ap.partition(indices, axis=axis)
-        # ensure axis with q-th is first
-        ap = np.moveaxis(ap, axis, 0)
-        axis = 0
-
-        # Check if the array contains any nan's
-        if np.issubdtype(a.dtype, np.inexact):
-            indices = indices[:-1]
-            n = np.isnan(ap[-1:, ...])
-
-        if zerod:
-            indices = indices[0]
-        r = take(ap, indices, axis=axis, out=out)
+    # ensure axis with q-th is first, as the qth axis is first in the output
+    ap = np.moveaxis(ap, axis, 0)
+    axis = 0
```

Comment:
```suggestion
    del axis
```
maybe? Its not set again, is it? It shows up once below in a `take`. This might be nice to move to the first `ap` definition, but not sure its better...

Reply:
I considered that. It was also tempting to leave the axis unmoved until the end.

Reply:
Done